flow/storage/objectstorage/objectstorage.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		return err
+	}
+	flowset = flowset.Filter(packetFilter)
+	rawpackets := make(map[string]*flow.RawPackets)
+	for _, currFlow := range flowset.Flows {
+		if fr, ok := rawpackets[currFlow.UUID]; ok {
+			fr.RawPackets = append(fr.RawPackets, currFlow.LastRawPackets)
+		} else {
+			linkType, err := currFlow.LinkType()
+			if err != nil {
+				linkType = layers.LinkTypeNull
+			}
+			rawpackets[currFlow.UUID] = &flow.RawPackets{
+				LinkType:   linkType,
+				RawPackets: []*flow.RawPacket{currFlow.LastRawPackets},
+			}
+		}
+	}
+	return rawpackets, nil
+}
+
+// SearchMetrics searches flow metrics matching filters in the database
+func (c *ObjectStorage) SearchMetrics(fsq filters.SearchQuery, metricFilter *filters.Filter) (map[string][]common.Metric, error) {
+	flowset, err := c.SearchFlows(fsq)
+	if err != nil {
+		return err
+	}
+	flowset = flowset.Filter(metricFilter)
+	metrics := make(map[string][]common.Metric)
+	for _, currFlow := range flowset.Flows {
+		metrics[currFlow.UUID] = append(metrics[currFlow.UUID], currFlow.Metric)
+	}
+	return metrics, nil
+}
+
+func (c *ObjectStorage) getAllFlows() (*flow.FlowSet, error) {
+	objectKeys, err := c.client.ListObjects(c.bucket, c.objectPrefix)
+	if err != nil {
+		logging.GetLogger().Error("Failed to list objects: ", err)
+		return nil, err
+	}
+
+	flowset := flow.NewFlowSet()
+	for _, objectKey := range objectKeys {
+		objectBytes, err := c.client.ReadObject(c.bucket, objectKey)
+		if err != nil {
+			logging.GetLogger().Error("Failed to read object: ", err)
+			return nil, err
+		}
+
+		var flows []*flow.Flow
+		err = json.Unmarshal(objectBytes, &flows)
+		if err != nil {
+			logging.GetLogger().Warn("Failed to JSON-decode object: ", err)
+		} else {
+			flowset.Flows = append(flowset.Flows, *flows)
+		}
+	}
+
+	return flowset, nil
+}
+
+// SearchFlows search flow matching filters in the database
+func (c *ObjectStorage) SearchFlows(fsq filters.SearchQuery) (*flow.FlowSet, error) {
+	flowset, err := c.getAllFlows()
+	if err != nil {
+		return err
+	}
+
+	return flowset.SearchQuery(fsq), nil
+}
+
+// Start the Object Storage client
+func (c *ObjectStorage) Start() {
+}
+
+// Stop the Object Storage client
+func (c *ObjectStorage) Stop() {
+}
+
+// New creates a new object storage backend client
+func New(backend string) *ObjectStorage {
+	path := "storage." + backend
+
+	endpoint := config.GetConfig().GetString(path + ".endpoint")
+	region := config.GetConfig().GetString(path + ".region")
+	bucket := config.GetConfig().GetString(path + ".bucket")
+	accessKey := config.GetConfig().GetString(path + ".access_key")
+	secretKey := config.GetConfig().GetString(path + ".secret_key")
+	entityID := config.GetConfig().GetString(path + ".entity_id")
+	collectorType := config.GetConfig().GetString(path + ".collector_type")
+	collectorID := strconv.Itoa(config.GetConfig().GetInt(path + ".collector_id"))
+	rootOutputDir := config.GetConfig().GetString(path + ".root_output_dir")
```

Comment:
I don't see why we need four parameters:
- entity_id
- collector_type
- collector_id
- root_output_dir

These all are used to construct the path leading to where the data is stored - so maybe just use one parameter instead:
- data_path

Reply:
added object_prefix parameter to replace all of these, as you suggested